tests/tests_e2e/ustream_aux_e2e/ustream_aux_e2e.c

```diff
@@ -210,6 +215,144 @@ TEST_FUNCTION(az_ustream_multi_read_concurrent_succeed)
+    size_t current_pos;
+    result = az_ustream_get_position(&ustream_instance, &current_pos);
+    ASSERT_ARE_EQUAL(int, AZ_ULIB_SUCCESS, result);
+    ASSERT_ARE_EQUAL(int, 0, current_pos);
+
+    result = az_ustream_get_position(&ustream_instance_split, &current_pos);
+    ASSERT_ARE_EQUAL(int, AZ_ULIB_SUCCESS, result);
+    ASSERT_ARE_EQUAL(int, split_location, current_pos);
+
+    result = az_ustream_read(&ustream_instance, buf_result, USTREAM_COMPLIANCE_EXPECTED_CONTENT_LENGTH, &returned_size);
+    ASSERT_ARE_EQUAL(int, AZ_ULIB_SUCCESS, result);
+    ASSERT_ARE_EQUAL(int, 0, strncmp((const char*)buf_result, (const char*)USTREAM_COMPLIANCE_LOCAL_EXPECTED_CONTENT_2,
+                                                       strlen((const char*)USTREAM_COMPLIANCE_LOCAL_EXPECTED_CONTENT_2)));
+
+    memset(&buf_result, 0, USTREAM_COMPLIANCE_EXPECTED_CONTENT_LENGTH);
+
+    result = az_ustream_read(&ustream_instance_split, buf_result, USTREAM_COMPLIANCE_EXPECTED_CONTENT_LENGTH, &returned_size);
+    ASSERT_ARE_EQUAL(int, AZ_ULIB_SUCCESS, result);
+    ASSERT_ARE_EQUAL(int, 0, strncmp((const char*)buf_result, (const char*)USTREAM_COMPLIANCE_LOCAL_EXPECTED_CONTENT_3,
+                                                       strlen((const char*)USTREAM_COMPLIANCE_LOCAL_EXPECTED_CONTENT_3)));
+
+    ///cleanup
+    az_ustream_dispose(&ustream_instance);
+    az_ustream_dispose(&ustream_instance_split);
+}
+
+TEST_FUNCTION(az_ustream_split_split_and_reset_succeed)
+{
+    ///arrange
+    AZ_USTREAM_DATA_CB* data_cb = (AZ_USTREAM_DATA_CB*)malloc(sizeof(AZ_USTREAM_DATA_CB));
+    AZ_USTREAM ustream_instance;
+    AZ_ULIB_RESULT result = az_ustream_init(&ustream_instance, data_cb, free, USTREAM_COMPLIANCE_LOCAL_EXPECTED_CONTENT_2_3,
+        strlen((const char*)USTREAM_COMPLIANCE_LOCAL_EXPECTED_CONTENT_2_3), NULL);
+    ASSERT_ARE_EQUAL(int, AZ_ULIB_SUCCESS, result);
+
+    uint8_t buf_result[USTREAM_COMPLIANCE_EXPECTED_CONTENT_LENGTH] = { 0 };
+    size_t split_location = strlen((const char*)USTREAM_COMPLIANCE_LOCAL_EXPECTED_CONTENT_2);
+    size_t returned_size;
+
+    AZ_USTREAM ustream_instance_split;
+
+    result = az_ustream_read(&ustream_instance, buf_result, 2, &returned_size);
+    ASSERT_ARE_EQUAL(int, AZ_ULIB_SUCCESS, result);
+    ASSERT_ARE_EQUAL(int, 2, returned_size);
+
+    ///act
+    result = az_ustream_split(&ustream_instance, &ustream_instance_split, split_location);
+    ASSERT_ARE_EQUAL(int, AZ_ULIB_SUCCESS, result);
+    result = az_ustream_reset(&ustream_instance);
+    ASSERT_ARE_EQUAL(int, AZ_ULIB_SUCCESS, result);
+
+    ///assert
+    size_t remaining_size;
+    result = az_ustream_get_remaining_size(&ustream_instance, &remaining_size);
+    ASSERT_ARE_EQUAL(int, AZ_ULIB_SUCCESS, result);
+    ASSERT_ARE_EQUAL(int, split_location, remaining_size);
+
+    ///cleanup
+    az_ustream_dispose(&ustream_instance);
+    az_ustream_dispose(&ustream_instance_split);
+}
+
+TEST_FUNCTION(az_ustream_split_split_and_concat_succeed)
```

Comment:
>az_ustream_split_split_and_concat_succeed [](start = 14, length = 41)

I'm missing the opposite of it, something like concatenate 3 uStreams, split the result in a half, and release the second part before release the first one.

Reply:
done

---
In reply to: [333766454](https://github.com/Azure/azure-ulib-c/pull/36#discussion_r333766454) [](ancestors = 333766454)

Reply:
added two more tests. first does the concat from three ustream and splits in two. checks both ensuring they are properly size and read the correct data. the third concats, splits, releases the second, and then reads properly the first

---
In reply to: [333778389](https://github.com/Azure/azure-ulib-c/pull/36#discussion_r333778389) [](ancestors = 333778389,333766454)